src/python/aqueduct_executor/operators/connectors/tabular/bigquery.py

```diff
@@ -14,26 +14,44 @@ def __init__(self, config: config.BigQueryConfig):
 
         credentials_info = json.loads(config.service_account_credentials)
         self.credentials = service_account.Credentials.from_service_account_info(credentials_info)
+        self.client = bigquery.Client(credentials=self.credentials, project=self.project_id)
 
     def authenticate(self) -> None:
-        # pandas_gbq does not have an explicit authenticate method, so we execute a test query
-        pandas_gbq.read_gbq("SELECT 1;", project_id=self.project_id, credentials=self.credentials)
+        # There is no explicit authenticate method, so we execute a test query
+        self.client.query("SELECT 1;")
 
     def discover(self) -> List[str]:
-        # TBD eng-708-investigate-how-to-list-tables-for-bigquery
-        return []
+        all_tables = []
+        for dataset in self.client.list_datasets():
+            tables = self.client.list_tables(dataset.dataset_id)
+            all_tables.extend([table.full_table_id.split(":")[-1] for table in tables])
+        return all_tables
 
     def extract(self, params: extract.RelationalParams) -> pd.DataFrame:
-        df = pandas_gbq.read_gbq(
-            params.query, project_id=self.project_id, credentials=self.credentials
-        )
+        query = self.client.query(params.query)
+        df = query.result().to_dataframe()
         return df
 
     def load(self, params: load.RelationalParams, df: pd.DataFrame) -> None:
-        pandas_gbq.to_gbq(
-            df,
-            params.table,
-            project_id=self.project_id,
-            if_exists=params.update_mode.value,
-            credentials=self.credentials,
+        update_mode = params.update_mode.value
```

Comment:
For some reason, it won't work unless I do `.value` because `common.UpdateMode.APPEND` is an object `<UpdateMode.APPEND: 'append'>` and when I write`'append' == common.UpdateMode.APPEND` it results in `False`.

Reply:
Oh wait I think I get what you mean--I need to remove .value from here and the if statements